internal/adminapi/client_test.go

```diff
@@ -0,0 +1,239 @@
+package adminapi
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"math/big"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+	"time"
+)
+
+var ca *x509.Certificate
+var caPEM *bytes.Buffer
+var caPrivateKeyPEM *bytes.Buffer
+var certPEM *bytes.Buffer
+var certPrivateKeyPEM *bytes.Buffer
+
+func TestMakeHTTPClient(t *testing.T) {
+
+	buildTLS(t)
+
+	var opts = HTTPClientOpts{
+		true,
+		"",
+		"",
+		caPEM.String(),
+		nil,
+		"",
+		certPEM.String(),
+		"",
+		certPrivateKeyPEM.String(),
+	}
```

Comment:
This change would improve clarity for code readers, but will compile the same.

```suggestion
	opts := HTTPClientOpts{
		TLSSkipVerify: false,
		TLSServerName: "",
		CACertPath: "",
		CACert: caPEM.String(),
		Headers: nil,
		TLSClientCertPath: "",
		TLSClientCert: certPEM.String(),
		TLSClientKeyPath: "",
		TLSClientKey: certPrivateKeyPEM.String(),
	}
```

This also changes to short variable declaration, which is optional, but shorter.

Additionally, this changes `TLSSkipVerify` from `true` to `false`, because the nature of what we're testing requires that we explicitly validate certificate trust.

Reply:
For the certificate check, I go the issue that I cannot register during my tests the generated self-signed root CA, thus leading for the connectivity test to fail if the TLS verification is activated. I understand that this is not ideal.
Do you have an idea if this could be fixed ?